src/commands.rs

```diff
@@ -3867,61 +3867,89 @@ fn cmd_git_push(
 
     let mut tx;
     let mut branch_updates = vec![];
-    if let Some(branch_name) = &args.branch {
-        if let Some(update) = branch_updates_for_push(
-            workspace_command.repo().as_repo_ref(),
-            &args.remote,
-            branch_name,
-        )? {
-            branch_updates.push((branch_name.clone(), update));
-        } else {
-            writeln!(
-                ui,
-                "Branch {}@{} already matches {}",
-                branch_name, &args.remote, branch_name
-            )?;
+    if !args.branch.is_empty() {
+        for branch_name in &args.branch {
+            if let Some(update) = branch_updates_for_push(
+                workspace_command.repo().as_repo_ref(),
+                &args.remote,
+                branch_name,
+            )? {
+                branch_updates.push((branch_name.clone(), update));
+            } else {
+                writeln!(
+                    ui,
+                    "Branch {}@{} already matches {}",
+                    branch_name, &args.remote, branch_name
+                )?;
+            }
         }
+        let branch_description = match &args.branch[..] {
+            [only_branch] => format!("branch {only_branch}"),
+            [.., last_branch] => {
+                format!(
+                    "branches {} and {last_branch}",
+                    &args.branch[..args.branch.len() - 1].join(", ")
+                )
+            }
+            _ => panic!("Assumed there is at least one branch"),
+        };
         tx = workspace_command.start_transaction(&format!(
-            "push branch {branch_name} to git remote {}",
+            "push {branch_description} to git remote {}",
```

Comment:
I think `branch.join(", ")` is good enough for transaction description, but if we need to handle singular/plural forms properly, `make_branch_term()` can be reused. `make_branch_term(branch_names)` could also be abstracted to `(singular, plural, items: IntoIterator<Item = AsRef<str>>)`.

Reply:
Yes, I was thinking of generalizing it. The generalization could also handle empty lists, restrictions on the length of the output or number of items to be output, etc.

But for now, make_branch_term is good.